pkg/common/telemetry/inmem_test.go

```diff
@@ -14,6 +21,92 @@ func TestNewInmemRunner(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestDefaultEnabledNewInmemRunner(t *testing.T) {
+	t.Run("enabled when block undeclared", func(t *testing.T) {
+		runner, err := newInmemRunner(testInmemConfig())
+		assert.Nil(t, err)
+		assert.True(t, runner.isConfigured())
+	})
+
+	t.Run("enabled flag undeclared", func(t *testing.T) {
+		config := testInmemConfig()
+		config.FileConfig = FileConfig{
+			InMem: &InMem{},
+		}
+		runner, err := newInmemRunner(config)
+		assert.Nil(t, err)
+		assert.True(t, runner.isConfigured())
+	})
+
+	t.Run("enabled flag declared", func(t *testing.T) {
+		enabledFlag := true
+
+		config := testInmemConfig()
+		config.FileConfig = FileConfig{
+			InMem: &InMem{
+				Enabled: &enabledFlag,
+			},
+		}
+		runner, err := newInmemRunner(config)
+		assert.Nil(t, err)
+		assert.True(t, runner.isConfigured())
+	})
+}
+
+func TestDisabledNewInmemRunner(t *testing.T) {
+	enabledFlag := false
+
+	config := &MetricsConfig{
+		ServiceName: "foo",
+		FileConfig: FileConfig{
+			InMem: &InMem{
+				Enabled: &enabledFlag,
+			},
+		},
+	}
+	runner, err := newInmemRunner(config)
+	assert.Nil(t, err)
+	assert.False(t, runner.isConfigured())
+}
+
+func TestWarnOnFutureDisable(t *testing.T) {
+	logger, hook := test.NewNullLogger()
+
+	// Get a real logrus.Entry
+	logger.SetLevel(logrus.DebugLevel)
+	logger.Debug("boo")
```

Comment:
This seems a little unreliable to emit here. If something else ends up logging while setting up the metrics stuff (e.g. due to a future refactor) then the check below will fail. I'd suggest emitting the sentinel message right before sending the signal.

Reply:
we need a log message to occur in order to have an Entry to pass to the sink config

Reply:
You can set `Logger` to the `logger` returned from `test.NewNullLogger()` (conforms to the `logrus.FieldLogger` interface).

Reply:
InMem wants an `Entry` in its setup (even though MetricsConfig takes a FieldLogger):

```
	if entry, ok := c.Logger.(*logrus.Entry); ok {
```

Reply:
Ah. I see. That code could be a little more permissive since `*logrus.Logger` also has a `Writer() *io.PipeWriter` method. I'll put that on my plate to fix.